system/Database/Forge.php

```diff
@@ -403,27 +403,38 @@ public function addField($field)
     /**
      * Add Foreign Key
      *
-     * @param string $fieldName
-     * @param string $tableName
-     * @param string $tableField
-     * @param string $onUpdate
-     * @param string $onDelete
-     *
-     * @return Forge
+     * @param string|string[] $fieldName
+     * @param string          $tableName
+     * @param string|string[] $tableField
+     * @param string          $onUpdate
+     * @param string          $onDelete
      *
      * @throws DatabaseException
+     *
+     * @return Forge
      */
-    public function addForeignKey(string $fieldName = '', string $tableName = '', string $tableField = '', string $onUpdate = '', string $onDelete = '')
+    public function addForeignKey($fieldName = '', string $tableName = '', $tableField = '', string $onUpdate = '', string $onDelete = '')
     {
-        if (! isset($this->fields[$fieldName])) {
-            throw new DatabaseException(lang('Database.fieldNotExists', [$fieldName]));
+        $fieldName  = (array) $fieldName;
+        $tableField = (array) $tableField;
+        $errorNames = [];
+
+        foreach ($fieldName as $name) {
+            if (! isset($this->fields[$name])) {
+                $errorNames[] = $name;
+            }
+        }
+
+        if ($errorNames !== []) {
+            throw new DatabaseException(lang('Database.fieldNotExists', $errorNames));
```

Comment:
If the count > 1, the array's first element will only be translated to the exception message. The others would be disregarded. I think you need to call implode here and adjust the message for singular and plural fields.

Reply:
Please add a test showing that adding unknown fields will throw an exception.